ipsw.sh

```diff
@@ -68,52 +68,48 @@ LLB_patch=$(buddy LLB:Patch)
 iBoot_patch=$(buddy iBoot:Patch)
 DeviceTree_patch=$(budd DeviceTree:Patch)
 
-FirmwarePath=$(/usr/libexec/PlistBuddy -c 'Print :All_flash\ Path' FirmwareBundles/$bundlename/Info.plist)
+FirmwarePath=$(/usr/libexec/PlistBuddy -c 'Print :All_flash\ Path' FirmwareBundles/"$bundlename"/Info.plist)
 
 echo "Unzipping Files"
 
-unzip $2 "${LLB}" >/dev/null
-unzip $2 "${iBoot}" >/dev/null
-unzip $2 "${DeviceTree}" >/dev/null
-unzip $2 "${manifest}" >/dev/null
+unzip "$2" "${LLB}" >/dev/null
+unzip "$2" "${iBoot}" >/dev/null
+unzip "$2" "${DeviceTree}" >/dev/null
+unzip "$2" "${manifest}" >/dev/null
 
 echo "Decrypting Files"
 
-./xpwntool ${LLB} ${LLB}.dec -iv ${LLB_iv} -k ${LLB_k} -decrypt >/dev/null
-./xpwntool ${iBoot} ${iBoot}.dec -iv ${iBoot_iv} -k ${iBoot_k} -decrypt >/dev/null
-./xpwntool ${DeviceTree} ${DeviceTree}.dec -iv ${DeviceTree_iv} -k ${DeviceTree_k} -decrypt >/dev/null
+./xpwntool "${LLB}" "${LLB}.dec" -iv "${LLB_iv}" -k "${LLB_k}" -decrypt >/dev/null
+./xpwntool "${iBoot}" "${iBoot}.dec" -iv "${iBoot_iv}" -k "${iBoot_k}" -decrypt >/dev/null
+./xpwntool "${DeviceTree}" "${DeviceTree}.dec" -iv "${DeviceTree_iv}" -k "${DeviceTree_k}" -decrypt >/dev/null
 
 echo "Removing original files"
 
-rm -f $LLB
-rm -f $iBoot
-rm -f $DeviceTree
+rm -f "$LLB" "$iBoot" "$DeviceTree"
 
 echo "Patching Files"
 
-bspatch ${LLB}.dec ${FirmwarePath}LLB.img3 ${LLB_patch}
-bspatch ${iBoot}.dec ${FirmwarePath}iBoot.img3 ${iBoot_patch}
-bspatch ${DeviceTree}.dec ${FirmwarePath}DeviceTree.img3 ${DeviceTree_patch}
+bspatch "${LLB}.dec" "${FirmwarePath}LLB.img3" "${LLB_patch}"
+bspatch "${iBoot}.dec" "${FirmwarePath}iBoot.img3" "${iBoot_patch}"
+bspatch "${DeviceTree}.dec" "${FirmwarePath}DeviceTree.img3" "${DeviceTree_patch}"
 
 echo "Editing manifest"
 
-echo "LLB.img3" >> ${manifest}
-echo "iBoot.img3" >> ${manifest}
-echo "DeviceTree.img3" >> ${manifest}
+echo "LLB.img3" "iBoot.img3" "DeviceTree.img3" >> "${manifest}"
 
 echo "Cleaning up"
 
-rm -f ${FirmwarePath}*.dec
+rm -f "${FirmwarePath}*.dec"
```

Comment:
There is no reason why this shouldn't work. Are you sure you are running it correctly?